### PR TITLE
feat(spans): Add a span count distribution endpoint

### PR DESCRIPTION
**Pull request by udameli**
For the performance issues POC, we want to show how a span count changed to illustrate the effect of the duplicate spans problem. To do that, I'm creating a new private endpoint that will return histogram data for a span count per transaction for a specific span op. This is an experimental feature while we're figuring out if it's valuable for the end product. I will follow up with more comprehensive tests once we're set on the usefulness of the count histogram 
image

### CODE REVIEW

**Comment by udameli** [comment]
there is a lot of histogram logic duplication. I didn't want to get into refactoring the existing span or transaction histogram endpoint because we're not set on the final solution yet

File: src/sentry/api/endpoints/organization_events_spans_count_histogram.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    query = serializers.CharField(required=False)
+    numBuckets = serializers.IntegerField(min_value=1, max_value=100)
+    precision = serializers.IntegerField(default=0, min_value=0, max_value=4)
+    min = serializers.FloatField(required=False)
+    max = serializers.FloatField(required=False)
+    dataFilter = serializers.ChoiceField(choices=DATA_FILTERS, required=False)
+
+    def validate(self, data):
+        if "min" in data and "max" in data and data["min"] > data["max"]:
+            raise serializers.ValidationError("min cannot be greater than max.")
```
Comment (wmak) [inline src/sentry/api/endpoints/organization_events_spans_count_histogram.py:24]: nit:
```suggestion
        if "min" in data and "max" in data and data["min"] >= data["max"]:
            raise serializers.ValidationError("min must be less than max.")
```

File: src/sentry/api/endpoints/organization_events_spans_count_histogram.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        return data
+
+    def validate_spanOp(self, spanOp: str):
+        if len(spanOp) < 1:
```
Comment (wmak) [inline src/sentry/api/endpoints/organization_events_spans_count_histogram.py:28]: nit:
```suggestion
        if spanOp == "":
```

**Review by wmak — APPROVED** [review]
2 nits, but LGTM

**Comment by wmak** [comment]
Oh just to confirm, there's a backlog card to cleanup the extra histogram code right?

**Comment by udameli** [comment]
> Oh just to confirm, there's a backlog card to cleanup the extra histogram code right?

there is one now :) created [PERF-1648](https://getsentry.atlassian.net/browse/PERF-1648) 👍

**Pull request merged**